### DISCUSSION TITLE
Pyarmor Pro Upgrade Error

### DISCUSSION THREAD

**Discussion by cedriclmao** [dashingsoft/pyarmor#1217] [Q&A]
Hi I asked five days ago and i got a answer but it didnt worked, so i wanted to ask again.

PS C:\Users\Cedric\Desktop\PYARMOR> pyarmor reg -u -p non-profits pyarmor-regcode-to-pro.txt
INFO     Python 3.9.4
INFO     Pyarmor 8.1.9 (trial), 000000, non-profits
INFO     Platform windows.x86_64

Pyarmor 8.0 changes EULA and uses new commands
It's different from previous Pyarmor totally
Please read this import notes first:
https://github.com/dashingsoft/pyarmor/issues/980
Do not upgrade to Pyarmor 8 if don't know what are changed

I have known the changes of Pyarmor 8? (yes/no/help) yes
INFO     prepare "pyarmor-regcode-to-pro.txt"
INFO     query key file from server
ERROR    HTTP Error 400
ERROR    missing old license code

**Comment by jondy** [comment]
How do you understand this error message `missing old license code`? And do you have an old license?

**Reply by cedriclmao** [reply]
I dont know if i have one and look into my old discussion please, i did there what you said and got another error

**Reply by jondy** [reply]
In a short, please check FAQ in the documentation to find how to refund this order, it doesn't work for you.

**Reply by cedriclmao** [reply]
Why doesent it work for me?
